book/go/api/writing-your-own-stateful-application.md

```diff
@@ -48,7 +48,7 @@ The first element, `rvt.GetVotes()`, is a message that we will send on to our ne
 
 ### State and StateBuilder
 
-We are going to partition by letter. We'll have a separate state object for every letter.
+We are going to partition by letter. Since we know the full set of keys in advance, it's a good idea to create them up front. If the application dynamically creates the keys instead, there will be a slight, but necessary, performance hit. We'll have a separate state object for every letter.
```

Comment:
`If the application dynamically creates state partitions for these keys instead, there will be a slight, but necessary, performance hit.`